src/main/java/io/sdkman/broker/download/CandidateDownloadHandler.java

```diff
@@ -39,15 +39,32 @@ public CandidateDownloadHandler(VersionRepo versionRepo, AuditRepo auditRepo, Do
     @Override
     public void handle(Context ctx) throws Exception {
         RequestDetails.of(ctx).ifPresentOrElse(details -> {
-            logger.info("Received download request for: " + details);
+            logger.info("Received download request for: {}", details);
             Platform.of(details.getPlatform())
                     .ifPresentOrElse(p -> versionRepo
                                     .fetch(details.getCandidate(), details.getVersion())
                                     .then((List<Version> downloads) ->
                                             downloadResolver.resolve(downloads, p.name())
                                                     .ifPresentOrElse(v -> {
-                                                        record(details, p.id(), v.getPlatform());
-                                                        ctx.redirect(302, v.getUrl());
+                                                        audit(details, p.id(), v.getPlatform());
+
+                                                        if (!details.isRenderChecksum()) {
```

Comment:
It feels wrong to be adding this to the `CandidateDownloadHandler`. The single responsibility of this handler is to provide candidate downloads. I think we should have an entirely new Handler class for checksums.